doc/reference/reference_lua/config/utils_schema.rst

```diff
@@ -0,0 +1,1163 @@
+            -   ``w.path`` -- the path to the schema node
+            -   ``w.error()`` -- a function for printing human-readable error messages
+
+        -   ``ctx`` -- additional *context* for the transformation function. Can be
+            used to provide values for a specific call.
+
+        An example of the transformation function:
+
+        .. code-block:: lua
+
+            local function f(data, w, ctx)
+                if w.schema.type == 'string' and data ~= nil then
+                    return data:gsub('{{ *foo *}}', ctx.foo)
+                end
+                return data
+            end
+
+        The ``map()`` method traverses all fields of the schema records,
+        even if they are ``nil`` or ``box.NULL`` in the provided configuration.
+        This allows using this method to set computed default values for missing
+        fields. Note that this is not the case for maps and arrays since the schema
+        doesn't define their fields to traverse.
+
+        :param any data: value at the given path
+        :param any f: walkthrough node, described below
+        :param any f_ctx: user-provided context for the transformation function
+
+        :return: transformed configuration data
+
+    ..  _config-schema_object-merge:
+
+    ..  method:: merge(data_a, data_b)
+
+        Merge two configurations. The method merges configurations in a single
+        node hierarchy, preferring the latter in case of a collision.
+
+        The following merge rules are used:
+
+        -   any present value is preferred over ``nil`` and ``box.NULL``
+        -   ``box.NULL`` is preferred over ``nil``
+        -   for scalar and array nodes, the right-hand value is used
+
+            .. important::
+
+                Note that the method doesn't concatenate arrays. Left hand array
+                items are discarded.
+
+        -   records and maps are deeply merged, that is, the merge is performed
+            recursively for their nested nodes
+
+        :param any a: configuration data
+        :param any b: configuration data
+
+        :return: merged configuration data
+
+    ..  _config-schema_object-pairs:
+
+    ..  method:: pairs()
+
+        Walk over the schema and return scalar, array and map schema nodes
+
+        .. important::
+
+            The method doesn't return record nodes.
+
+        :return: a luafun iterator
+
+        **Example:**
+
+        .. code-block:: lua
+
+            for _, w in schema:pairs() do
+                local path = w.path
+                local schema = w.schema
+                -- <...>
+            end
+
+    ..  _config-schema_object-set:
+
+    ..  method:: set(data, path, value)
+
+        Set a given value at the given path in a configuration.
+        The path can be either a dot-separated string (``http.scheme``) or
+        an array-like table (``{ 'http', 'scheme'}``).
+
+        :param any data: configuration data
+        :param any path: path to the target node
```

Comment:
I wouldn't say that `path` has any type. It is either a string in the dot-notation or an array-like table of strings.

Reply:
Fixed